src/internal/hooks/use-component-metrics/index.ts

```diff
@@ -0,0 +1,22 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useEffect } from 'react';
+
+import { ComponentMetrics } from '../../analytics';
+
+export function useComponentMetrics(
+  componentName: string,
+  elementRef: React.RefObject<HTMLElement>,
+  getDetails: () => Record<string, string | boolean | number | undefined>
+) {
+  useEffect(() => {
+    if (!elementRef.current) {
```

Comment:
Just a curios question, why we check for the ref when is not needed by the `ComponentMetrics.componentMounte(...)` at all?

Reply:
Originally I was going to something similar to the [performance hook](https://github.com/cloudscape-design/components/blob/main/src/internal/hooks/use-performance-marks/index.ts#L50-L57) to check if the element was visible on page. I opted for just firing the event regardless and ended up leaving this behind as a bit of a defensive check that this should only run once it's been attached in the dom. It could possibly be removed though.